### ISSUE TITLE
🐛 BUG: Order of params for routing has to match

### ISSUE THREAD

**Issue by p13rnd** [withastro/astro#2235]
### What version of `astro` are you using?

0.21.13

### What package manager are you using?

npm

### What operating system are you using?

Windows

### Describe the Bug

I am working on my routes and page structure, while doing this @RafidMuhymin noticed that the order of input params is important in the frontmatter script.

So this code:

```js
---
export async function getStaticPaths() {
  const data = await fetch(
    `${import.meta.env.PUBLIC_CONTENT_URL}posts/posts.json`
  ).then((response) => response.json());
  console.log(data);
  return data.posts.map((post) => {
    return {
      params: { category: post.category, slug: post.slug },
    };
  });
}
const { slug } = Astro.request.params;
console.log(slug);
---
<p>Post: {slug}</p>
```

does not work if category and slug switch. 

```js
params: { category: post.category, slug: post.slug }, // works
params: { slug: post.slug, category: post.category }, // doesn't work
```

Folder structure is /pages/posts/[category]/[slug].astro

Link to the regarding discord thread: https://discord.com/channels/830184174198718474/845451724738265138/922115198414901278

Cheers,
Pi

### Link to Minimal Reproducible Example

https://github.com/p13rnd/arcturus/blob/master/src/pages/posts/%5Bcategory%5D/%5Bslug%5D.astro

**Comment by jonathantneal** [comment]
Hello! I was able to get the example started, but I’m unable to test this without an ENV value for `PUBLIC_CONTENT_URL`. Would you be able to give me one, or let me know of a suitable fallback?

**Comment by p13rnd** [comment]
Hi @jonathantneal 
this is just a mock server running on my local which returns a json:

```json
{
  "posts": [
    {
      "slug": "testpost-1",
      "category": "test123"
    },
    {
      "slug": "test-post-2",
      "category": "test123"
    }
  ]
}
```

Is that suitable as well? Let me know any further details I can provide.

Cheers
Pi

**Comment by tony-sull** [comment]
Confirmed, the property order in a path's `params` has to match the folder/file structure or the route won't be found

**Comment by tony-sull** [comment]
Just merged [PR #2593](https://github.com/withastro/astro/pull/2593) with the fix and test coverage, thanks for catching this bug @p13rnd!
